### ISSUE TITLE
Readonly function arguments should be declared as such

### ISSUE THREAD

**Issue by GustavTaxen** [lucaong/minisearch#172]
IMO, the methods and functions in the public API for MiniSearch should make use of TypeScript's ability to declare that arguments are immutable. For example, in MiniSearch.ts (line 530), we have

```
 addAll (documents: T[]): void { ... }
 ```
 
I'm assuming that this method does not mutate the input documents, so it'd be great if the function definition could say so explicitly, i.e.,

```
 addAll (documents: ReadonlyArray<T>): void { ... }
 ```

In addition to providing the information, it also avoids TypeScript from complaining when you try to pass an immutable array to this method.

**Comment by lucaong** [comment]
Hi @GustavTaxen , thanks for the suggestion. I will explore this possibility, which sounds reasonable to me.

**Comment by lucaong** [comment]
Tackled in #184

**Comment by lucaong** [comment]
This is released on NPM as `v6.0.0-beta.1` (and will soon be `v6.0.0`). Thank you for the suggestion.
